decode.go

```diff
@@ -1239,64 +1262,97 @@ func (d *decoder) parseToTag(v reflect.Value) error {
 	return nil
 }
 
-func (d *decoder) parseToTime() (tm time.Time, err error) {
-	t := d.nextCBORType()
-
+// parseToTime decodes the current data item as a time.Time. The bool return value is false if and
+// only if the destination value should remain unmodified.
+func (d *decoder) parseToTime() (time.Time, bool, error) {
 	// Verify that tag number or absence of tag number is acceptable to specified timeTag.
-	if t == cborTypeTag {
+	if t := d.nextCBORType(); t == cborTypeTag {
 		if d.dm.timeTag == DecTagIgnored {
-			// Skip tag number
+			// Skip all enclosing tags
 			for t == cborTypeTag {
 				d.getHead()
 				t = d.nextCBORType()
 			}
+			if d.nextCBORNil() {
+				return time.Time{}, false, nil
+			}
```

Comment:
We need to skip CBOR nil here before returning. 

```suggestion
			if d.nextCBORNil() {
			        d.skip() // skip CBOR nil
				return time.Time{}, false, nil
			}
```